pkg/cloud/services/vpc/vpc.go

```diff
@@ -54,4 +54,5 @@ type Vpc interface {
 	CreateLoadBalancerPoolMember(options *vpcv1.CreateLoadBalancerPoolMemberOptions) (*vpcv1.LoadBalancerPoolMember, *core.DetailedResponse, error)
 	DeleteLoadBalancerPoolMember(options *vpcv1.DeleteLoadBalancerPoolMemberOptions) (*core.DetailedResponse, error)
 	ListLoadBalancerPoolMembers(options *vpcv1.ListLoadBalancerPoolMembersOptions) (*vpcv1.LoadBalancerPoolMemberCollection, *core.DetailedResponse, error)
+	GetAccountID() string
```

Comment:
Provider ID format is: ibm://<account_id>///<cluster_name>/<instance_id> so we need to fetch the account id of user, Did not get a any other way to fetch it so added a method here in the VPC interface.

Reply:
most of the functions are straight forward for generating the mock code here, wondering if there is a better way to deal with this instead of adding it here

cc @Prajyot-Parab @dharaneeshvrd

Reply:
This can be moved out to the utils package.

Reply:
We create a function like this in utils package
```
func GetAccountID() (string, error) {
	auth, err := authenticator.GetAuthenticator()
	if err != nil {
		return "", err
	}
	accountID, err := GetAccount(auth)
	return accountID, err
}
```

But by this way authenticator.GetAuthenticator() will be called twice for each reconciliation loop one at the beginning while initializing the services and one here.